internal/provider/vaultradar/resource_radar_source_github_enterprise.go

```diff
@@ -0,0 +1,235 @@
+	"github.com/hashicorp/terraform-provider-hcp/internal/clients"
+	"github.com/hashicorp/terraform-provider-hcp/internal/provider/modifiers"
+)
+
+var (
+	_ resource.Resource              = &sourceGitHubEnterpriseResource{}
+	_ resource.ResourceWithConfigure = &sourceGitHubEnterpriseResource{}
+)
+
+func NewSourceGitHubEnterpriseResource() resource.Resource {
+	return &sourceGitHubEnterpriseResource{}
+}
+
+type sourceGitHubEnterpriseResource struct {
+	client *clients.Client
+}
+
+func (r *sourceGitHubEnterpriseResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_vault_radar_source_github_enterprise"
+}
+
+func (r *sourceGitHubEnterpriseResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "This Vault Radar resource manages HCP Cloud scans of GitHub Enterprise Server sources.",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Computed:    true,
+				Description: "The ID of this resource.",
+			},
+			"domain_name": schema.StringAttribute{
+				Description: "Fully qualified domain name of the server. (Example: myserver.acme.com)",
+				Required:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+				Validators: []validator.String{
+					stringvalidator.RegexMatches(regexp.MustCompile(`^\S+$`),
+						"must not contain white space",
+					),
+				},
+			},
+			"github_organization": schema.StringAttribute{
+				Description: `GitHub organization Vault Radar will monitor. Example: "octocat" for the org https://yourcodeserver.com/octocat`,
+				Required:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+				Validators: []validator.String{
+					stringvalidator.RegexMatches(regexp.MustCompile(`^[a-zA-Z0-9-_.]+$`),
+						"must contain only letters, numbers, hyphens, underscores, or periods",
+					),
+				},
+			},
+			"token": schema.StringAttribute{
+				Description: "GitHub personal access token.",
+				Required:    true,
+				Sensitive:   true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			// Optional inputs
+			"project_id": schema.StringAttribute{
+				Description: "The ID of the HCP project where Vault Radar is located. If not specified, the project specified in the HCP Provider config block will be used, if configured.",
+				Optional:    true,
+				Computed:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+		},
+	}
+}
+
+type gitHubEnterpriseSource struct {
+	ID                 types.String `tfsdk:"id"`
+	DomainName         types.String `tfsdk:"domain_name"`
+	GitHubOrganization types.String `tfsdk:"github_organization"`
```

Comment:
nit: Can we remove `GitHub` prefix since it is already a part of `gitHubEnterpriseSource`?

Reply:
I don't want this to be confused with the HCP Organization.